AnyText/AnyText.Lsp/LspServer.SemanticToken.cs

```diff
@@ -0,0 +1,221 @@
+                        }
+                    }
+                };
+
+            // If no changes, return an empty delta
+            return new SemanticTokensDelta
+            {
+                ResultId = Guid.NewGuid().ToString(),
+                Edits = Array.Empty<SemanticTokensEdit>()
+            };
+        }
+
+        /// <summary>
+        ///     Handles the <c>textDoocument/semanticTokens/range</c> request from the client. This is used to retrieve semantic
+        ///     tokens within a specific range in the document.
+        /// </summary>
+        /// <param name="arg">The JSON token containing the parameters of the range request. (SemanticTokensRangeParams)</param>
+        /// <returns>A <see cref="SemanticTokens" /> object containing the semantic tokens within the specified range.</returns>
+        public SemanticTokens QuerySemanticTokensRange(JToken arg)
+        {
+            var semanticTokensRangeParams = arg.ToObject<SemanticTokensRangeParams>();
+            var uri = semanticTokensRangeParams.TextDocument.Uri;
+            var range = semanticTokensRangeParams.Range;
+
+            if (!_documents.TryGetValue(uri, out var document))
+                return new SemanticTokens { ResultId = null, Data = Array.Empty<uint>() };
+
+
+            var tokens = document.GetSemanticElementsFromRoot(AsParsePosition(range.Start), AsParsePosition(range.End))
+                .ToArray();
+
+            return new SemanticTokens
+            {
+                ResultId = Guid.NewGuid().ToString(),
+                Data = tokens
+            };
+        }
+
+        /// <summary>
+        ///     Creates the registration for semantic token requests.
+        /// </summary>
+        /// <param name="languageId">The language identifier for the document.</param>
+        /// <param name="parser">The parser used to extract token types and modifiers.</param>
+        /// <returns>A <see cref="Registration" /> object with semantic token registration options.</returns>
+        private Registration CreateSemanticTokenRegistration(string languageId, Parser parser)
+        {
+            var registrationOptions = new SemanticTokensRegistrationOptions
+            {
+                DocumentSelector = new[]
+                {
+                    new DocumentFilter
+                    {
+                        Language = languageId
+                    }
+                },
+                Full = new { delta = true },
+                Range = true,
+                Legend = new SemanticTokensLegend
+                {
+                    tokenTypes = parser.Context.Grammar.TokenTypes,
+                    tokenModifiers = parser.Context.Grammar.TokenModifiers
+                }
+            };
+            return new Registration
+            {
+                RegisterOptions = registrationOptions,
+                Id = Guid.NewGuid().ToString(),
+                Method = "textDocument/semanticTokens"
```

Comment:
Da gibt es bei LSPtypes bestimmt eine Konstante für, oder nicht?

Reply:
In LSPTypes gibt es keine Konstante dafür. Habe die Konstante in eine neue statische Klasse verlegt.